_posts/2022-06-15-inline-assembly-memory-side-effects-bug.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: post
+published: true
+title: 'Optimizer Bug regarding Memory Side Effects of Inline Assembly'
+date: '2022-06-15'
+author: Solidity Team
+category: Security Alerts
+---
+
+On June 5 2022, John Toman of the Certora development team reported an optimizer bug
+that can cause memory writes in inline assembly blocks to be incorrectly removed
+under certain conditions.
+
+The bug was introduced in Solidity 0.8.13 with a new Yul optimizer step meant to
+remove unused stores to memory and storage.
+
+We assigned the bug a severity of "medium".
+
+## Which Contracts are Affected?
+
+The Yul optimizer considers all memory writes in the outermost Yul block that are
+never read from as unused and removes them. This is valid when that Yul block is
+the entire Yul program, which is always the case for the Yul code generated by the
+new via-IR pipeline. Inline assembly blocks are never optimized in isolation when
+using that pipeline. Instead they are optimized as a part of the whole Yul input.
+
+However, the legacy code generation pipeline (which is still the default) runs the
+Yul optimizer individually on an inline assembly block if the block does not refer
+to any local variables defined in the surrounding Solidity code. Consequently, memory
+writes in such inline assembly blocks are removed as well, if the written memory is
+never read from in the same assembly block, even if the written memory is accessed
+later, for example by a subsequent inline assembly block.
+
+Fortunately, the fact that the legacy code generation pipeline does not run the Yul
+optimizer at all on inline assembly blocks that do access local Solidity variables,
+reduces the number of affected cases significantly. Most inline assembly blocks either
+read or write values from or to local variables of the surrounding Solidity code, or
+are entirely self-contained, or take over the program flow up until to the end of
+the transaction. Thereby, the impact of the bug is likely low in practice. However,
+since the behaviour in affected cases is thoroughly unexpected, we assigned a severity
```

Comment:
"unexpected" sounds a bit mild for this case. How about this?
```suggestion
since the consequences in affected cases can be severe, we assigned a severity
```